mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/memory/MemoryDeletedMessagesVault.java

```diff
@@ -41,10 +47,16 @@
 import reactor.core.publisher.Mono;
 
 public class MemoryDeletedMessagesVault implements DeletedMessageVault {
+    private final RetentionConfiguration retentionConfiguration;
     private final Table<User, MessageId, Pair<DeletedMessage, byte[]>> table;
-
-    public MemoryDeletedMessagesVault() {
-        table = HashBasedTable.create();
+    private final Clock clock;
+    private DeleteByQueryExecutor deleteByQueryExecutor;
+
+    public MemoryDeletedMessagesVault(RetentionConfiguration retentionConfiguration, Clock clock) {
+        this.deleteByQueryExecutor = new DeleteByQueryExecutor(this);
```

Comment:
are we having cyclic dependencies between DeletedMessageVault & DeleteByQueryExecutor? is that okay?

Reply:
Yes that is Okay